docs/services/ml/upload-training-script.md

```diff
@@ -10,48 +10,231 @@ no_service: true
+   labels_filename = "labels.txt"
+   unknown_label = "UNKNOWN"
+
+   API_KEY = os.environ['API_KEY']
+   API_KEY_ID = os.environ['API_KEY_ID']
+
+   # This parses the required args for the training script.
+   # The model_dir variable will contain the output directory where
+   # the ML model that this script creates should be stored.
+   # The data_json variable will contain the metadata for the dataset
+   # that you should use to train the model.
+   def parse_args():
+       """Dataset file and model output directory are required parameters. These must be parsed as command line
+           arguments and then used as the model input and output, respectively.
+       """
+       parser = argparse.ArgumentParser()
+       parser.add_argument("--dataset_file", dest="data_json", type=str)
+       parser.add_argument("--model_output_directory", dest="model_dir", type=str)
+       args = parser.parse_args()
+       return args.data_json, args.model_dir
+
+   # This is used for parsing the dataset file (produced and stored in Viam),
+   # parse it to get the label annotations
+   # Used for training classifiction models
+   def parse_filenames_and_labels_from_json(
+       filename: str, all_labels: ty.List[str], model_type: str
+   ) -> ty.Tuple[ty.List[str], ty.List[str]]:
+       """Load and parse JSON file to return image filenames and corresponding labels.
+       Args:
+           filename: JSONLines file containing filenames and labels
+           all_labels: list of all N_LABELS
+           model_type: string single_label or multi_label
+       """
+       image_filenames = []
+       image_labels = []
+
+       with open(filename, "rb") as f:
+           for line in f:
+               json_line = json.loads(line)
+               image_filenames.append(json_line["image_path"])
+
+               annotations = json_line["classification_annotations"]
+               labels = [unknown_label]
+               for annotation in annotations:
+                   if model_type == multi_label:
+                       if annotation["annotation_label"] in all_labels:
+                           labels.append(annotation["annotation_label"])
+                   if model_type == single_label:
+                       if annotation["annotation_label"] in all_labels:
+                           labels = [annotation["annotation_label"]]
+               image_labels.append(labels)
+       return image_filenames, image_labels
+
+
+   # Parse the dataset file (produced and stored in Viam) to get
+   # bounding box annotations
+   # Used for training object detection models
+   def parse_filenames_and_bboxes_from_json(
+       filename: str,
+       all_labels: ty.List[str],
+   ) -> ty.Tuple[ty.List[str], ty.List[str], ty.List[ty.List[float]]]:
+       """Load and parse JSON file to return image filenames
+       and corresponding labels with bboxes.
+       Args:
+           filename: JSONLines file containing filenames and bboxes
```

Comment:
This is my bad but args should all_labels: list of all N_LABELS here as well :)

Reply:
could you maybe suggest changes? I don't really understand what you want me to do. 
It's this button
![Screenshot 2024-08-12 at 18 19 44](https://github.com/user-attachments/assets/df36aede-e6db-463e-a096-34d4d8eba453)